### PR TITLE
Rename org.antlr.v4.test.tool.Java-LR.g4 to JavaLR.g4

### PR DESCRIPTION
**Pull request by HSorensen**
Eclipse AntlrIDE reports a syntax error in the tool-testsuite/test directroy for one file Java-LR.g4.

Rename the file from Java-LR.g4 to JavaLR.g4 and adjust the grammar statement accordingly.

### CODE REVIEW

**Comment by parrt** [comment]
Hi Can you add yourself to contributors.txt if not done and can you alter the code for this change too?

```
TestPerformance.java:     * left-recursive syntax (Java-LR.g4). {@code false} to use the standard
TestPerformance.java:        String sourceName = leftRecursive ? "Java-LR.g4" : "Java.g4";
```

Thanks!

**Comment by HSorensen** [comment]
I did run a complete test before pushing so I am wondering if some tests missing for the JavaLR grammar?

File: tool-testsuite/test/org/antlr/v4/test/tool/TestPerformance.java [hunk 1]
```diff
@@ -415,9 +415,9 @@ public void compileJdk() throws IOException, InterruptedException, ExecutionExce
 		assertTrue("The JDK_SOURCE_ROOT environment variable must be set for performance testing.", jdkSourceRoot != null && !jdkSourceRoot.isEmpty());
 
         compileJavaParser(USE_LR_GRAMMAR);
-		final String lexerName = "JavaLexer";
```
Comment (sharwell) [inline tool-testsuite/test/org/antlr/v4/test/tool/TestPerformance.java]: :exclamation: Please revert the style changes here. They don't follow the code formatting options used by this project.

File: tool-testsuite/test/org/antlr/v4/test/tool/TestPerformance.java [hunk 2]
```diff
@@ -1108,8 +1108,10 @@ private static long sum(long[] array) {
 	}
 
     protected void compileJavaParser(boolean leftRecursive) throws IOException {
-        String grammarFileName = "Java.g4";
-        String sourceName = leftRecursive ? "Java-LR.g4" : "Java.g4";
+        String grammarFileName = leftRecursive ? "JavaLR.g4"    : "Java.g4";
```
Comment (sharwell) [inline tool-testsuite/test/org/antlr/v4/test/tool/TestPerformance.java]: :bulb: There is no need to have separate `grammarFileName` and `sourceName` variables anymore. You can use `grammarFileName` for both.

File: .gitignore [hunk 3]
```diff
@@ -53,3 +53,4 @@ bilder.pyc
 bild.log
 
 bild_output.txt
+/bin/
```
Comment (sharwell) [inline .gitignore]: :bulb: consider setting this in **.git/info/exclude** instead, since the `bin` directory is not created by this project's build.

**Comment by sharwell** [comment]
@HSorensen This test is excluded from the tests run by the Maven build. You'll need to enable it locally by removing [this line](https://github.com/HSorensen/antlr4/blob/master/tool-testsuite/test/org/antlr/v4/test/tool/TestPerformance.java#L412).

**Comment by HSorensen** [comment]
Just a question regarding the preferred workflow for pull-requests?
Do I close the pull request and you merge later, or does the request remain open until you decide to merge?
The reason for asking is I want to start take-two to proper support include source into the lexer-scan and as long as this pull request is open new commits is added.

**Comment by sharwell** [comment]
@HSorensen You should leave the pull request open until it is merged. You can avoid the issue of new commits getting added by creating a new branch - I personally never do new work on my **master** branch which helps avoid these types of problems. If you create a new branch for your next proposal you can start working on it immediately. :+1:

**Comment by HSorensen** [comment]
@sharwell Just to clarify a bit more. Since the include stuff will touch some of the core of antlr do you prefer to have a pull request open showing the WIP and you can comment as development progresses or do you prefer to have full functioning and complete code in the pull request?

**Comment by HSorensen** [comment]
I see there are some conflicts now. Should I prepare a new pull requests with resolved conflicts?

**Comment by parrt** [comment]
Hi. maybe close this, create a branch, pull in master, add your changes, create new PR?

**Comment by HSorensen** [comment]
Yeah, makes sense.

**Pull request closed without merging**